lib/mapRoutes.js

```diff
@@ -177,3 +194,21 @@ module.exports = function (app) {
 
   return app
 }
+
+function validatePrefix (urlPrefix) {
+  // remove whitespace
+  if (typeof urlPrefix === 'string') {
+    urlPrefix = urlPrefix.replace(/\s/g, '')
+    // test for valid urlPrefix
+    if (/^[0-9a-zA-Z-_.+!*'(),/$]+$/.test(urlPrefix)) {
+      if (urlPrefix.charAt(0) !== '/') {
+        urlPrefix = [urlPrefix.slice(0, 0), '/', urlPrefix.slice(0)].join('')
```

Comment:
can i assume these changes are not breaking? for instance, if i have an app with a bunch of routes that are already predefined like `/foo/bar` and then i set `urlPrefix` to `/baz`, can i assume that `/foo/bar` becomes `/baz/foo/bar` now, but if i reset `urlPrefix` back to `/` the route still works as originally defined? (`/foo/bar`)

we won't get weird stuff going on like `//foo/bar` right?

Reply:
This case is handled because anytime `'/'` is the prefix we don't setup a separate router, we use the built in router within `app`. The only issue that someone using roosevelt could potentially run into is the path to the images, stylesheets, or any client-side js files being relative to the route. 
Example:`urlPrefix:  '/baz'` would have the stylesheets located at `/baz/css/*.css` but if the route is `/baz/foo/bar` it might render the view with the path to the stylesheet in `/baz/foo/bar/css/*.css`.
They would be required to prepend the `urlPrefix` by passing it through the global model and to the their views which could get messy. I haven't really figured out a better way to overcome this issue.